docs/docs/running-hello-world.md

```diff
@@ -73,216 +75,235 @@ The web server loads the enclave and waits for requests from the client. Amongst
+
+1. The host loads the enclave, which generates a remote attestation report.
+2. The client connects to the host and downloads the attestation report.
+3. The client uses the attestation report to verify that it is connected to the enclave it expected, and that the
+   enclave is running in the expected mode.
+4. The client encrypts the "reverse-me" string using the public key provided in the attestation.
+5. In the message to the enclave, the client also includes a public key for the enclave to use when responding.
+6. The client sends the encrypted message to the host's REST API.
+7. The host passes the encrypted message to the enclave.
+8. The enclave decrypts the client's message, reverses the string, and encrypts the response using the client's
    public key.
-9. The enclave passed the encrypted message to the host, was sent back to the client as an HTTP response to the same
+9. The enclave passes the encrypted message to the host, and is sent back to the client as an HTTP response to the same
    request.
 
-Congratulations on running your first enclave!
-
 ### Enclave Constraint
-The `--constraint` parameter of the client command-line interface defines the 
-properties of an acceptable enclave. This includes the signing key 
+The `--constraint` parameter of the client defines the properties of an acceptable enclave.
+This includes the signing key, which is also present in the host output in the previous section:
 ```
 S:0000000000000000000000000000000000000000000000000000000000000000
 ```
-which we saw in the host output in the previous
-section.
 
-Try rerunning the command with a different code signer:
+Try rerunning the command with a different signing key:
 ```
-"S:2222222222222222222222222222222222222222222222222222222222222222 PROD:1 SEC:INSECURE"
+java -jar client/build/libs/client.jar "S:2222222222222222222222222222222222222222222222222222222222222222 PROD:1
+SEC:INSECURE" reverse-me
 ```
 
-You should see the following error:
+You will get the following error, as the client's signing key(code signer) does not match the enclave's signing key:
 > com.r3.conclave.common.InvalidEnclaveException: Enclave code signer does not match any of the acceptable code signers.
 (key hash 0000000000000000000000000000000000000000000000000000000000000000 vs acceptable
 2222222222222222222222222222222222222222222222222222222222222222)
 
-For more information on how to choose a constraint, see [Enclave constraints](constraints.md).
-
+For more information on choosing a constraint, see [Enclave constraints](constraints.md).
 
 ## Beyond mock mode
-Eventually you will want to run your application in one of the other [enclave modes](enclave-modes.md), namely simulation,
-debug, or release. This section describes how to build and run the sample in these other modes.
+This section describes how to build and run the sample in other [enclave modes](enclave-modes.md), namely
+simulation, debug, and release.
```

Comment:
I think this is the first time we do mention enclave modes. Mode _debug_ is not used in the rest of the doc. I would add it along the simulation one to highlight it a bit more - see suggestion below (or as I commented above, maybe we can completely move these sections to a new doc).

Reply:
Done.

Reply:
Can we have a quick sentence/addition that says that only release mode is secure and has SGX guarantees. I'm sure this is mention in enclave-modes.md, but if modes are being introduced here, then the most important mode of them all needs to be highlighted. Essentially, we don't want developers thinking that all the modes are created equal.